### ISSUE TITLE
Problem adding second account for v.3.4.5 and not dot.com domain after reboot

### ISSUE THREAD

**Issue by somar05** [custom-components/alexa_media_player#1082]
**Describe the bug**
It looks that two differents accounts doesn't works fine, at least with non .com domain. The first account looks working fine (account added, reboot to test,...), but when I add the second one only works meanwhile I don't restart Home Assistant. After that, a few errors appears and relogin is required but with not success.

**To Reproduce**
1. Delete previous integration version
2. Add account according procedure
3. Test, reboot, test (all ok)
4. Add second account and looks fine but..
5. Relogin is required after a few minutes. No matters if reboot or not.

**System details**
 - Home-assistant (version): 2020.12.2
 - Hassio (Yes/No): yes
 - alexa_media  (version from `const.py` or HA startup): 3.4.5
 - alexapy (version from `pip show alexapy` or HA startup): 
 - Amazon [2FA is enabled](https://github.com/custom-components/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account) (y/n). We will not debug login issues if unanswered:  yes

**Logs**

> Logger: alexapy.alexawebsocket
Source: /usr/local/lib/python3.8/site-packages/alexapy/alexawebsocket.py:109
First occurred: 10:08:44 (1 occurrences)
Last logged: 10:08:44
Websocket is missing ubid-main and ubid-acbes cookies; please report this if anything isn't working.

> Logger: alexapy.helpers
Source: /usr/local/lib/python3.8/site-packages/alexapy/helpers.py:153
First occurred: 10:04:26 (7 occurrences)
Last logged: 10:09:15
alexaapi.get_device_preferences((<alexapy.alexalogin.AlexaLogin object at 0x6a316a48>,), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',)
alexaapi.get_devices((<alexapy.alexalogin.AlexaLogin object at 0x6a316a48>,), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',)
alexaapi.get_dnd_state((<alexapy.alexalogin.AlexaLogin object at 0x6a316a48>,), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',)
alexaapi.get_authentication((<alexapy.alexalogin.AlexaLogin object at 0x6a316a48>,), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',)
alexaapi.get_activities((<alexapy.alexalogin.AlexaLogin object at 0x6a316a48>, 10), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',)

**Additional context**
Domain .es
Rolling back to 3.4.1 works fine

**Comment by alandtse** [comment]
WIth the new oauth we don't support two logins of the same account to different domains. When we get the cookies using oauth to change domains, it will invalidate all prior cookies.  Is there a reason you are doing that?

**Comment by somar05** [comment]
No @alandtse, my scenario is two differents Amazon accounts in the same domain (amazon.es)

**Comment by Hightower3004** [comment]
i have the same problem (amazon.de)

**Comment by Hightower3004** [comment]
Logger: alexapy.alexalogin
Source: /usr/local/lib/python3.8/site-packages/alexapy/alexalogin.py:1159
First occurred: 2. Januar 2021, 23:00:07 (9 occurrences)
Last logged: 11:11:34

Forgot password page detected; Amazon has detected too many failed logins. Please check to see if Amazon requires any further action. You may have to wait before retrying.


But the password is right. My Apps (Amazon / Alexa) is works. Only HA is not working.

**Comment by rtirelli** [comment]
I think the best way to fix every bug is to uninstall the integration.

**Comment by umrath** [comment]
Same here:
2 different amazon.de accounts fail with 3.4.5, no matter what I try. I keep getting the re-authentication questions every few minutes - and that fails then.

Downgraded to 3.4.1 and it works just fine.

**Comment by user514** [comment]
...and same here. Two different amazon.es accounts and reauthentications asked continuosly. Enter info and get a windows titled 'Aborted', but the text inside the windows says its ok authenticated :-/
-minutes later, 2 notifications to reauthenticate.

**Comment by umrath** [comment]
Installed 3.4.7 - and it throws errors for me. The devices show up as "not available".
Downgrade to 3.4.1 does not cure the error anymore. They remain unavailable.

---

Logger: homeassistant.config_entries
Source: custom_components/alexa_media/__init__.py:259 
First occurred: 10:01:49 (2 occurrences) 
Last logged: 10:01:49

Error setting up entry ***redacted***@posteo.de - amazon.de for alexa_media
Error setting up entry ***redacted***@posteo.de - amazon.de for alexa_media
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/connector.py", line 999, in _create_direct_connection
    hosts = await asyncio.shield(
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/connector.py", line 868, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/resolver.py", line 31, in resolve
    infos = await self._loop.getaddrinfo(
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 825, in getaddrinfo
    return await self.run_in_executor(
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/alexapy/helpers.py", line 141, in wrapper
    return await func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/alexapy/alexalogin.py", line 509, in login
    if await self.test_loggedin(cookies):
  File "/usr/local/lib/python3.8/site-packages/alexapy/alexalogin.py", line 405, in test_loggedin
    get_resp = await self._session.get(
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/client.py", line 513, in _request
    conn = await self._connector.connect(
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/connector.py", line 543, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/connector.py", line 895, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/usr/local/lib/python3.8/site-packages/alexapy/aiohttp/connector.py", line 1005, in _create_direct_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
alexapy.aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host alexa.amazon.de:443 ssl:default [Try again]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 236, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/alexa_media/__init__.py", line 259, in async_setup_entry
    await login.login(cookies=await login.load_cookie())
  File "/usr/local/lib/python3.8/site-packages/alexapy/helpers.py", line 151, in wrapper
    raise AlexapyConnectionError from ex
alexapy.errors.AlexapyConnectionError

**Comment by alandtse** [comment]
Check your network. Its saying it can't connect to Amazon. Also open a new issue so I have enough info to assist.

**Comment by umrath** [comment]
I already checked the network and fail to find any issues.

**Comment by umrath** [comment]
Looks like the network is fine - but homeassistant is not.
For some reason the name resolution within home assistant is not working properly. Investigating.

I will raise a new ticket if necesary.

**Comment by umrath** [comment]
Alexa Media Player >3.4.1 is still failing for me. Tested with 3.4.8 yesterday and the days before.

I still do get rather frequent information that the authentication needs to get recreated.
Sometimes it seems to do that on its own. So, I enter the integrations page - and I see no error.
Sometimes I need to manually confirm what has already been configured.
Most of the time the integration has at least some outages when this happens.

I downgraded to 3.4.1 without changing any configuration or even re-authenticating and it works just fine again for >24 hours.
